ksqldb-functional-tests/src/test/resources/query-validation-tests/source-stream.json

```diff
@@ -66,6 +66,27 @@
         "type": "io.confluent.ksql.util.KsqlException",
         "message": "Cannot delete topic for read-only source: INPUT"
       }
+    },
+    {
+      "name": "rejects CREATE OR REPLACE SOURCE ... statements",
+      "statements": [
+        "CREATE OR REPLACE SOURCE STREAM INPUT (K STRING KEY, text STRING) WITH (kafka_topic='test_topic', value_format='DELIMITED');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Cannot add stream 'INPUT': CREATE OR REPLACE is not supported on source streams or tables."
+      }
+    },
+    {
+      "name": "rejects CREATE OR REPLACE ... statements on existing source streams",
+      "statements": [
+        "CREATE SOURCE STREAM INPUT (K STRING KEY, text STRING) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE OR REPLACE STREAM INPUT (K STRING KEY, text STRING) WITH (kafka_topic='test_topic', value_format='DELIMITED');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Cannot add stream 'INPUT': CREATE OR REPLACE is not supported on source streams or tables."
+      }
```

Comment:
What about the third case: `CREATE STREAM INPUT` followed by `CREATE OR REPLACE SOURCE STREAM INPUT`

Reply:
Done